components/layout_2020/fragment_tree/box_fragment.rs

```diff
@@ -192,6 +192,38 @@ impl BoxFragment {
         )
     }
 
+    pub fn overflow_clip_rect(&self, containing_block_rect: &PhysicalRect<Au>) -> PhysicalRect<Au> {
+        let mut clip_rect = self
+            .padding_rect()
+            .translate(containing_block_rect.origin.to_vector());
+
+        // Adjust by the overflow clip margin.
+        // https://drafts.csswg.org/css-overflow/#overflow-clip-margin
+        let clip_margin = Au::from(self.style.get_margin().overflow_clip_margin);
+        clip_rect = clip_rect.inflate(clip_margin, clip_margin);
+
+        let mut overflow = self
+            .scrollable_overflow()
+            .translate(containing_block_rect.origin.to_vector());
+
+        let overflow_style = self.style.effective_overflow();
+        if overflow_style.x == ComputedOverflow::Clip {
+            let min_x = overflow.min_x().max(clip_rect.min_x());
+            let max_x = overflow.max_x().min(clip_rect.max_x());
+            overflow.origin.x = min_x;
+            overflow.size.width = (max_x - min_x).max(Au::zero());
```

Comment:
Super-nit to avoid a pair of parenthesis:

```suggestion
            overflow.size.width = Au::zero().max(max_x - min_x);
```

Reply:
Resolve outdated conversations.